src/ast/types/ast.ts

```diff
@@ -1,23 +1,21 @@
 import { CompilationUnit } from "./packages-and-modules";
 import {
-  BinaryExpression,
+  Block,
+  BlockStatement,
+  Expression,
+  ExpressionName,
   Literal,
   LocalVariableDeclarationStatement,
-  UnaryExpression,
 } from "./blocks-and-statements";
 
-export interface ExpressionMap {
-  BinaryExpression: BinaryExpression;
-  Literal: Literal;
-  UnaryExpression: UnaryExpression;
-}
-
-type Expression = ExpressionMap[keyof ExpressionMap];
-
 interface NodeMap {
   CompilationUnit: CompilationUnit;
   LocalVariableDeclarationStatement: LocalVariableDeclarationStatement;
+  Block: Block;
+  BlockStatement: BlockStatement;
   Expression: Expression;
+  ExpresssionName: ExpressionName;
+  Literal: Literal;
```

Comment:
Hm do we still need these? Since they are under `Expression` already

Reply:
I overlooked this, removed. In this case, shall we also remove `Block` and `LocalVariableDeclarationStatement` since they will be part of `BlockStatement`?